react.component/src/app/SearchBar/SearchBar.tsx

```diff
@@ -1,95 +1,61 @@
-import { ChangeEvent, Component, createRef, RefObject } from 'react';
+import { ChangeEvent, KeyboardEvent, MutableRefObject, useContext, useEffect, useRef, useState } from 'react';
 import './searchbar.scss';
-import { cards, CardT } from '../cards';
 import { ReactComponent as Lupa } from '../../assets/lupa.svg';
-
-interface SearchBarProps {
-  cards: CardT[];
-  updateData: (cards: CardT[]) => void;
-}
-
-export class SearchBar extends Component<SearchBarProps> {
-  wrapper: RefObject<HTMLDivElement>;
-  input: RefObject<HTMLInputElement> | undefined;
-
-  constructor(props: SearchBarProps) {
-    super(props);
-    this.state = { cards: this.props.cards };
-    this.handleChange = this.handleChange.bind(this);
-    this.wrapper = createRef<HTMLDivElement>();
-    this.input = createRef<HTMLInputElement>();
-  }
-
-  componentWillUnmount() {
-    const neededValue = this.input?.current?.value;
-    if (neededValue) {
-      localStorage.setItem('bestbookstore-input-data', neededValue.toString());
-    } else {
-      localStorage.setItem('bestbookstore-input-data', '');
-    }
-  }
-
-  componentDidMount(): void {
-    const myvalue = localStorage.getItem('bestbookstore-input-data');
-    if (myvalue && this.input && this.input.current) {
-      this.input.current.value = myvalue;
-
-      const filtered = cards.filter((card) =>
-      Object.values(card).find(
-        (value: string | number) =>
-          value.toString().toLowerCase().search(myvalue.toLowerCase()) !== -1
-      )
-    );
-    this.setState({ cards: filtered });
-    this.props.updateData(filtered);
+import { AppContext } from '../../context';
+import { BookType } from '../types';
+
+export const SearchBar = () => {
+  const { docs, isLoading, setIsLoading, getDocs, setDocs } = useContext(AppContext);
+
+  const wrapper: MutableRefObject<HTMLDivElement | null> = useRef(null);
+  const input: MutableRefObject<HTMLInputElement | null> = useRef(null);
```

Comment:
please remove typings - they are correctly infered from the `useRef` if you will use it as a generic type.

const input = useRef<HTMLInputElement | null>(null)